src/vellum/workflows/nodes/bases/base.py

```diff
@@ -43,7 +43,15 @@ def __new__(mcs, name: str, bases: Tuple[Type, ...], dct: Dict[str, Any]) -> Any
         # TODO: Inherit the inner Output classes from every base class.
         #   https://app.shortcut.com/vellum/story/4007/support-auto-inheriting-parent-node-outputs
 
-        if "Outputs" not in dct:
+        if "Outputs" in dct:
+            outputs_class = dct["Outputs"]
+            if not any(issubclass(base, BaseOutputs) for base in outputs_class.__bases__):
+                dct["Outputs"] = type(
+                    f"{name}.Outputs",
+                    (BaseOutputs,) + outputs_class.__bases__,  # Add BaseOutputs as first base
```

Comment:
[nit] we should likely filter out this `<class 'object'>` from these bases as it'll be redundant with `BaseOutputs` already being a subclass of it
![Screenshot 2025-02-09 at 12 10 33 PM](https://github.com/user-attachments/assets/c0638125-4e83-4a67-beda-fa9b9f045c04)